python-package/lightgbm/sklearn.py

```diff
@@ -956,7 +960,7 @@ def objective_(self) -> Union[str, _LGBM_ScikitCustomObjectiveFunction]:
         """:obj:`str` or :obj:`callable`: The concrete objective used while fitting this model."""
         if not self.__sklearn_is_fitted__():
             raise LGBMNotFittedError('No objective found. Need to call fit beforehand.')
-        return self._objective
+        return self._objective  # type: ignore[return-value]
```

Comment:
This is the same as what's described in #5788 .

`self._objective` is initialized to `None`.

https://github.com/microsoft/LightGBM/blob/0957ab72f7e46649dc0e48daf15d1c2a9381c6a3/python-package/lightgbm/sklearn.py#L402

And then overwritten with a string like `"regression"` or a custom objective function during training.

But `mypy` doesn't understand that it'll always be non-`None` if `self.__sklearn_is_fitted__()` is `True` (in other words, that it's not possible under normal operation for a user to ever call `model.objective_` and get `None` back).